app/presenters/api/alerts_index_presenter.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+class Api::AlertsIndexPresenter
+  include ::Draper::ViewHelpers
+
+  def initialize(raw_alerts:, params:, service:, current_account:)
+    @raw_alerts = raw_alerts
+    @params = params
+    @service = service
+    @pagination_params = { page: params[:page] || 1, per_page: params[:per_page] || 20 }
+    @sorting_params = [params[:sort], params[:direction]]
+
+    do_account_weird_stuff(current_account)
+  end
+
+  attr_reader :raw_alerts, :params, :service, :pagination_params, :sorting_params
+
+  delegate :total_entries, to: :alerts
+
+  def alerts
+    @alerts ||= raw_alerts.order_by(*sorting_params)
+                          .scope_search(search)
+                          .paginate(pagination_params)
+  end
+
+  def empty_state?
+    raw_alerts.empty?
+  end
+
+  def empty_search?
+    alerts.empty?
+  end
+
+  def toolbar_props # rubocop:disable Metrics/MethodLength
+    {
+      totalEntries: total_entries,
+      actions: [],
+      overflow: [{
+        href: h.polymorphic_path([:all_read, :admin, service, :alerts]),
+        label: t('.read_all_button.label'),
+        variant: :primary,
+        'data-method': :put,
+        'data-confirm': t('.read_all_button.confirm'),
+        isShared: true
+      }, {
+        href: h.polymorphic_path([:purge, :admin, service, :alerts]),
+        label: t('.purge_button.label'),
+        variant: :danger,
+        'data-method': :delete,
+        'data-confirm': t('.purge_button.confirm'),
+        isShared: true
+      }],
+      search: {
+        name: 'search[account][query]',
+        placeholder: t('.search_placeholder')
+      }
+    }
+  end
+
+  def page_title
+    service ? t('.page_title.service') : t('.page_title.all')
+  end
+
+  def link_to_account_for_alert(alert)
+    if (account = alert.cinstance.try(:user_account))
+      h.link_to account.org_name, h.admin_buyers_account_path(account)
+    else
+      h.tag.span('missing')
+    end
+  end
+
+  private
+
+  def search
+    # default to account_id and cinstance_id params if no search hash is passed
+    search_params = params.fetch(:search) { params.slice(:account_id, :cinstance_id) }
+
+    @search ||= ThreeScale::Search.new(search_params)
+  end
+
+  # TODO: this method looks very fishy, verify it is valid and useful.
+  def do_account_weird_stuff(current_account)
```

Comment:
I've been checking, and it looks like it **is** in fact required - without it search by account doesn't work.

So the thing is that alerts do not have their own index in sphinx, but we want to allow searching by buyer account, so apparently it works like this:

1. The controller receives a query by account name, e.g. `{"account"=>{"query"=>"aqw"}}`
2. As account name means nothing for alerts, it transforms this query in to the list of `account_id`s, this is done using a Sphinx search. This is what this method does.
Some logs:
```
[search] Allowed scopes: ["query", "state", "country_id", "plan_id", "created_within"]
[search] Selected {"query"=>"aqw"} from {"query"=>"aqw"}
  Sphinx Query (1.1ms)  SELECT * FROM `account_core` WHERE MATCH('*aqw*') AND `tenant_id` = 2 AND `sphinx_deleted` = 0 LIMIT 0, 1000000 OPTION cutoff=0, field_weights=(name=2)
  Sphinx  Found 1 results
```

3. So, now that the search scope has `account_id`s, we can query alerts - it doesn't use sphinx, but rather will use ActiveRecord's default `by_account_id`.
Releavant log:
```
[search] Allowed scopes: ["cinstance_id", "account_id", "timestamp", "level"]
[search] Selected {"account_id"=>[164]} from {"account"=>{"query"=>"aqw"}, "account_id"=>[164], "query"=>nil}
```
`164` is the ID of the buyer account that was found by query "aqw".

So, the method name should be changed to something meaningful :grimacing:

Reply:
Wow, OK! Good finding!!!